src/Spryker/Zed/SynchronizationBehavior/Persistence/Propel/Behavior/SynchronizationBehavior.php

```diff
@@ -301,10 +317,88 @@ protected function setGeneratedKey()
     }
 
     /**
-     * @param string $script
+     * @throws \Spryker\Zed\SynchronizationBehavior\Persistence\Propel\Behavior\Exception\MissingAttributeException
      *
      * @return string
      */
+    protected function addGenerateMappingResourceKeyMethod()
+    {
+        $parameters = $this->getParameters();
+        if (!isset($parameters['mapping_resource'])) {
+            return '/**
+ * @return void
+ */
+protected function setMappingResourceGeneratedKey()
+{
+}';
+        }
+        $keySuffix = null;
+        $storeSetStatement = '';
+        $localeSetStatement = '';
+        $referenceSetStatement = '';
+        $checkSuffixStatement = '';
+
+        if (!isset($parameters['resource']['value'])) {
+            throw new MissingAttributeException(sprintf(static::ERROR_MISSING_RESOURCE_PARAMETER, $this->getTable()->getPhpName()));
+        }
+
+        if (!isset($parameters['mapping_resource']['value'])) {
+            throw new MissingAttributeException(sprintf(static::ERROR_MISSING_MAPPING_RESOURCE_PARAMETER, $this->getTable()->getPhpName()));
+        }
+
+        $mappingResourceSuffix = "'{$parameters['mapping_resource']['value']}'";
+
+        $resource = $parameters['resource']['value'];
+
+        if (isset($parameters['mapping_resource_key_suffix_column'])) {
+            $filter = new UnderscoreToCamelCase();
+            $keySuffix = sprintf('get%s()', $filter->filter($parameters['mapping_resource_key_suffix_column']['value']));
+            $checkSuffixStatement = "if (empty(\$this->$keySuffix)) {
+         return;       
+    }
+            ";
+        }
+
+        if (isset($parameters['store'])) {
```

Comment:
I think we can extract these expressions, now it's repeated 2 times, same for locale